test/src/unit/template.test.js

```diff
@@ -53,705 +55,519 @@ const mockRepos = {
+                this.timeout(10000);
+                const templateController = new Templates('');
+                templateController.projectTemplatesNeedsRefresh = false;
+                templateController.projectTemplates = [
+                    {
+                        name: 'project1',
+                        projectStyle: 'otherprojectstyle',
+                    },
+                    {
+                        name: 'project2',
+                        projectStyle: 'projectstyle',
+                    },
                 ];
+                const templateList = await templateController.getTemplates(false, 'projectstyle');
+                templateList[0].name.should.equal('project2');
+                templateList.length.should.equal(1);
             });
-            it('returns only the default templates', async function() {
-                const output = await templateController.getAllTemplates();
-                output.should.deep.equal(defaultCodewindTemplates);
-            });
-        });
-        describe('and add a provider providing a valid template repo', function() {
-            let templateController;
-            before(async() => {
-                fs.ensureDirSync(testWorkspaceConfigDir);
-                templateController = new Templates(testWorkspaceDir);
-                await templateController.writeRepositoryList();
-                templateController.addProvider('valid repo', {
-                    getRepositories() {
-                        return [sampleRepos.appsody];
-                    },
+            describe('Verifys the Codewind default templates are there and all others contain the required keys', function() {
+                it('returns the default templates', async function() {
+                    this.timeout(30000);
+                    const templateController = new Templates('');
+                    const output = await templateController.getTemplates(false, false);
+                    output.should.contain.deep.members(defaultCodewindTemplates);
+                    for (let i = 0; i < output.length; i++) {
+                        const element = output[i];
+                        // List of keys required are generated from the API docs
+                        element.should.contain.keys('label', 'description', 'language', 'url', 'projectType');
+                    }
                 });
             });
-            after(() => {
-                fs.removeSync(testWorkspaceDir);
-            });
-            it('returns the default templates and more', async function() {
-                this.timeout = 10000;
-                const output = await templateController.getAllTemplates();
-                output.should.include.deep.members(defaultCodewindTemplates);
-                (output.length).should.be.above(defaultCodewindTemplates.length);
+            describe('and add an extra bad template repo', function() {
```

Comment:
'extra bad' sounds like it's a really bad template, could be `additional invalid template`?

Reply:
I'd also get rid of the and from this, and the below describe